src/TreeView/TreeView.tsx

```diff
@@ -417,7 +434,7 @@ const LeadingVisual: React.FC<TreeViewVisualProps> = props => {
   const children = typeof props.children === 'function' ? props.children({isExpanded}) : props.children
   return (
     <Slot name="LeadingVisual">
-      <Box sx={{color: 'fg.muted'}}>{children}</Box>
+      <Box sx={{display: 'flex', color: 'fg.muted'}}>{children}</Box>
```

Comment:
What do we need flexbox for? Is it just a vertical alignment thing? Or are we actually laying out multiple children?

Reply:
When you render an SVG (Octicon) in the leading visual slot, it adds unwanted extra space unless the container is a flex container.

Reply:
Ok, I figured it was something like that. I've dealt with that before, and I don't understand _why_ it works 🙈 

I think it also works for `grid`